AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/Mutations.dfy

```diff
@@ -137,32 +173,48 @@ module {:options "/functionSyntax:4" } Mutations {
         }
     }
 
-    if (
-        && !success?
-        && item.Type.ActiveHierarchicalSymmetricVersion?
-      ) {
-      var error := MutationErrorRefinement.VerifyActiveException(
+    if (!success?) {
+      var error := BuildErrorForFailure(
+        item,
+        throwAwayError,
+        localOperation,
+        kmsOperation
+      );
+      return Failure(error);
+    }
+
+    assert success?;
+    return Success(ActiveVerificationHolder.NotDecrypt());
+  }
+
+  method BuildErrorForFailure(
```

Comment:
Praise: Big fan of small methods.
However, this method is only good for the Active, right?
And, Build Error For Failure is not an efficient description,
as a Error and Failure are synonyms. 
Therefore:
 ```suggestion
  method BuildErrorForInvalidActive(
```

Reply:
Its also for DecryptOnly key. `BuildErrorForInvalidActiveAndDecryptOnly` could be better

Reply:
I am confused... OH... Because this is used to Verify Terminal?
Then `BuildVerificationError`.
In Mutations, we verify:
- the original active item
- every terminal item encountered

Thus, verification errors MUST fall into one of those two categories...
UNLESS we use also use the parent method (`VerifyEncryptedHierarchicalKey`)
to decrypt HV-2 items on the mutation path...
in which case... 
we have a confusion on paths,
because the....

Reply:
Let's do this:
- Name the method `BuildVerificationError`

And add a TODO to add a pre-condition that the `ActiveHierarchicalSymmetricVersion` Item's have the original context,
and the `HierarchicalSymmetricVersion` have the terminal context.

Add a TODO because refactoring to expose that information here maybe annoying
and more work than it is worth...

Reply:
Done at [88b357e](https://github.com/aws/aws-cryptographic-material-providers-library/pull/1442/commits/88b357e7e659de58a5faa91e3e797622a6196831)